app/services/association_authorizer.rb

```diff
@@ -1,20 +1,21 @@
 class AssociationAuthorizer
-  def self.authorized_associations(associations,  klass_name = nil, should_raise_exception = true)
+  def self.authorized_associations(associations,  klass_name = nil, should_raise_exception = true, action = 'view')
```

Comment:
How's action going to be overridden, since it's called from core code usually?  Or are you calling it directly somehow?